### ISSUE TITLE
Subflow re-dialing?

### ISSUE THREAD

**Issue by bardiir** [benjojo/bondcat#5]
Wouldn’t it make sense to add retry intervals to the subflows or even better in general do an interval to rediscover new paths?

As of now as far as I can see once a subflow is disconnected due to a network interruption it’s gone for good which means once every connection that was established originally failed once the overall connection fails even if there would be a new path available or an old path available again.

**Comment by benjojo** [comment]
Yeah there is a desire to do this, bgp.tools uses the same multipath library for it's relay nodes, and so I need to:

(A) port the fixes from the bgp.tools repo back here
(B) add auto subflow redial

I took a stab at this a few months ago and realised it was not just a single days worth of effort, but since it's highly desired, I will do it for bgp.tools and port it back here.

**Comment by bardiir** [comment]
I’d offer to help but sadly I have basically zero Go knowledge and it didn’t look like just add an interval and drop new dialers into the pool from what I could make sense of the code.
